utils/utils.go

```diff
@@ -497,9 +499,12 @@ type DeployableArtifactDetails struct {
 	TargetRepository string `json:"targetRepository,omitempty"`
 }
 
-func (detailes *DeployableArtifactDetails) CreateFileTransferDetails(rtUrl, targetRepository string) FileTransferDetails {
-	targetPath := rtUrl + targetRepository + "/" + detailes.ArtifactDest
-	return FileTransferDetails{SourcePath: detailes.SourcePath, TargetPath: targetPath, Sha256: detailes.Sha256}
+func (details *DeployableArtifactDetails) CreateFileTransferDetails(rtUrl, targetRepository string) (FileTransferDetails, error) {
+	url, err := url.Parse(rtUrl + targetRepository)
```

Comment:
We should and return this error here, if it is not nil.
I suggest we also explain why we're parsing the URL.